app/src/ui/app.tsx

```diff
@@ -1018,6 +1018,10 @@ export class App extends React.Component<IAppProps, IAppState> {
           popup.showDiscardChangesSetting === undefined
             ? true
             : popup.showDiscardChangesSetting
+        const discardingAllChanges =
```

Comment:
perhaps this should have a default value too?

Reply:
Unfortunately, after looking into it, this one may not be possible. The variable this one depends on is defined in the `app-state.ts` and when I switch it to the `discardingAllChanges: boolean = true` style it warns me that an initializer like this is only possible in a constructor or function implementation.

I will implement the other changes for now as it does not seem I can progress on this one.

Reply:
oops must have missed some context on this one!